api/class2.js

```diff
@@ -1,79 +1,202 @@
+  req,
+  res
+) {
+  //var newUpdate = `select * from customers where id =  ${req.params.id}`;
+  let newData = req.params;
+  var sql = ` UPDATE customers set  first_name = '${newData.first_name}', 
+  surname = '${newData.surname}'
+  where id = '${req.params.id}' `;
+
+  console.log(sql);
+  db.run(sql, [], (err, rows) => {
+    if (err) {
+      console.log("ERROR fetching from the database:", err);
+      return;
+    }
+    console.log("Request succeeded, new data fetched", rows);
+    res.status(200).json({
+      result_message: "Successfuly inserted new customer."
+    });
+  });
 });
 
-
 // get '/reservations'
 // TODO: add code here
-
+router.get("/reservations", function(req, res) {
+  var sql = "select * from reservations";
+  db.all(sql, [], (err, rows) => {
+    res.status(200).json({
+      reservations: rows
+    });
+  });
+});
 
 // get '/reservations/:id'
 // TODO: add code here
-
+router.get("/reservations/:id", function(req, res) {
+  var sql = `select * from reservations where id =  ${req.params.id}`;
+  db.all(sql, [], (err, rows) => {
+    if (err) {
+      console.log("ERROR fetching from the database:", err);
+      return;
+    }
+    console.log("Request succeeded, new data fetched", rows);
+    res.status(200).json({
+      customers: rows
+    });
+  });
+});
 
 // delete '/reservations/:id'
 // TODO: add code here
-
+router.delete("/reservations/:id", function(req, res) {
+  //var newUpdate = `select * from customers where id =  ${req.params.id}`;
+  var sql = ` DELETE from reservations where id = ${req.params.id} `;
+  console.log(sql);
+  db.run(sql, [], (err, rows) => {
+    if (err) {
+      console.log("ERROR fetching from the database:", err);
+      return;
+    }
+    console.log("Request succeeded, new data fetched", rows);
+    res.status(200).json({
+      result_message: "Successfuly inserted new customer."
+    });
+  });
+});
 
 // get '/reservations/starting-on/:startDate'
-// TODO: add code here
-
+router.get("/reservations/starting-on/:check_in_date", function(req, res) {
+  console.log(req.params.check_in_date);
+  var sql = `select * from reservations where check_in_date = '${
+    req.params.check_in_date
+  }'`;
+  db.all(sql, [], (err, rows) => {
+    if (err) {
+      console.log("ERROR fetching from the database:", err);
+      return;
+    }
+    console.log("Request succeeded, new data fetched", rows);
+    res.status(200).json({
+      customers: rows
+    });
+  });
+});
 
 // get '/reservations/active-on/:date'
-// TODO: add code here
 
+//COMMENTS - On this exercise i tried to make sure both the date and room_id has values as without room_id the customerwould not have stayed in the hotel for that date. As I could not think of any other criteria to compare it with. Please do let me know if I would have used something else. On schema.sql, I made sure some check_in_date null to see if it could work.
+
+router.get("/reservations/active-on/:check_in_date", function(req, res) {
```

Comment:
this should be /reservations/active-on/:date

you need to check if date is after check in date but before check out date.